src/transformers/trainer.py

```diff
@@ -1363,13 +1362,6 @@ def prediction_loop(
             prediction_loss_only if prediction_loss_only is not None else self.args.prediction_loss_only
         )
 
-        assert not getattr(
-            self.model.config, "output_attentions", False
-        ), "The prediction loop does not work with `output_attentions=True`."
-        assert not getattr(
-            self.model.config, "output_hidden_states", False
-        ), "The prediction loop does not work with `output_hidden_states=True`."
-
```

Comment:
Wouldn’t we want to put these lines inside an if statement? The prediction loop still doesn’t work with these outputs right?

Reply:
Nope it does now since the functions that detach/concat etc. all work on nested list/tuples of tensors :-)